scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/constraints/TimeRangeExpression.java

```diff
@@ -30,91 +30,78 @@ public class TimeRangeExpression {
    * @param domain x
    * @return x
    */
-  public Windows computeRange(Plan plan, Windows domain) {
+  public Windows computeRange(final SimulationResults simulationResults, final Plan plan, final Windows domain) {
 
     Windows inter = new Windows(domain);
-    //particularly important for combining with constant states as they are by definition adjacent values
-    //TODO:remove
-    //inter.doNotMergeAdjacent();
 
-    if (constantWin != null) {
+    if (constantWin != null && !inter.isEmpty()) {
       inter.intersectWith(constantWin);
     }
 
-    if (actTemplate != null) {
+    if (actTemplate != null && !inter.isEmpty()) {
       Windows actTw = new Windows();
-      //particularly important for combining with constant states as they are by definition adjacent values
-      //TODO:marker to remove
-      //actTw.doNotMergeAdjacent();
-      var minTimepoint = domain.minTimePoint();
-      var maxTimepoint = domain.maxTimePoint();
-      if(minTimepoint.isPresent()&& maxTimepoint.isPresent()) {
+      var minTimepoint = inter.minTimePoint();
+      var maxTimepoint = inter.maxTimePoint();
+      if(minTimepoint.isPresent() && maxTimepoint.isPresent()) {
         final var anchorActSearch = new ActivityExpression.Builder()
             .basedOn(actTemplate)
             .startsIn(Window.between(
                 Duration.max(Duration.ZERO, minTimepoint.get()),
                 Window.Inclusivity.Inclusive,
                 maxTimepoint.get(),
                 Window.Inclusivity.Exclusive)).build();
-        final var anchorActs = plan.find(anchorActSearch);
+        final var anchorActs = plan.find(anchorActSearch, simulationResults);
         for (var anchorAct : anchorActs) {
           var endInclusivity = Window.Inclusivity.Exclusive;
           if(anchorAct.getDuration().isZero()){
             endInclusivity = Window.Inclusivity.Inclusive;
           }
           actTw.add(Window.between(anchorAct.getStartTime(), Window.Inclusivity.Inclusive, anchorAct.getEndTime(), endInclusivity));
         }
-        inter = actTw;
       }
+      inter.intersectWith(actTw);
     }
 
     for (var otherExpr : timeRangeExpressions) {
-      Windows windowsState = otherExpr.computeRange(plan, domain);
+      if(inter.isEmpty()) break;
+      Windows windowsState = otherExpr.computeRange(simulationResults, plan, domain);
       inter.intersectWith(windowsState);
-      if (inter.isEmpty()) {
-        break;
-      }
     }
 
     for (var expr : stateExpr) {
-      Windows windowsState = expr.findWindows(plan, domain);
+      if(inter.isEmpty()) break;
+      final var domainOfInter = Window.between(inter.minTimePoint().get(), inter.maxTimePoint().get());
+      Windows windowsState = expr.evaluate(simulationResults, domainOfInter, null);
```

Comment:
The "empty" case of environment is an empty Map
```suggestion
      Windows windowsState = expr.evaluate(simulationResults, domainOfInter, Map.of());
```